code/game/gamemodes/shadowling/shadowling.dm

```diff
@@ -0,0 +1,319 @@
+		sleep(10)
+		shadow.current << "<br>"
+		shadow.current << "<span class='shadowling'><b><font size=3>You are a shadowling!</font></b></span>"
+		greet_shadow(shadow)
+		finalize_shadowling(shadow)
+		process_shadow_objectives(shadow)
+		//give_shadowling_abilities(shadow)
+	..()
+	return
+
+/datum/game_mode/proc/greet_shadow(datum/mind/shadow)
+	shadow.current << "<b>Currently, you are disguised as an employee aboard [station_name()]].</b>"
+	shadow.current << "<b>In your limited state, you have three abilities: Enthrall, Hatch, and Hivemind Commune.</b>"
+	shadow.current << "<b>Any other shadowlings are your allies. You must assist them as they shall assist you.</b>"
+	shadow.current << "<b>If you are new to shadowling, or want to read about abilities, check the wiki page at https://tgstation13.org/wiki/Shadowling</b><br>"
+
+
+/datum/game_mode/proc/process_shadow_objectives(datum/mind/shadow_mind)
+	var/objective = "enthrall" //may be devour later, but for now it seems murderbone-y
+
+	if(objective == "enthrall")
+		objective_explanation = "Ascend to your true form by use of the Ascendance ability. This may only be used with [required_thralls] collective thralls, while hatched, and is unlocked with the Collective Mind ability."
+		shadow_objectives += "enthrall"
+		shadow_mind.memory += "<b>Objective #1</b>: [objective_explanation]"
+		shadow_mind.current << "<b>Objective #1</b>: [objective_explanation]<br>"
+
+
+/datum/game_mode/proc/finalize_shadowling(datum/mind/shadow_mind)
+	var/mob/living/carbon/human/S = shadow_mind.current
+	shadow_mind.AddSpell(new /obj/effect/proc_holder/spell/self/shadowling_hatch(null))
+	shadow_mind.AddSpell(new /obj/effect/proc_holder/spell/self/shadowling_hivemind(null))
+	spawn(0)
+		update_shadow_icons_added(shadow_mind)
+		if(shadow_mind.assigned_role == "Clown")
+			S << "<span class='notice'>Your alien nature has allowed you to overcome your clownishness.</span>"
+			S.dna.remove_mutation(CLOWNMUT)
+
+/datum/game_mode/proc/add_thrall(datum/mind/new_thrall_mind)
+	if(!istype(new_thrall_mind))
+		return 0
+	if(!(new_thrall_mind in thralls))
+		update_shadow_icons_added(new_thrall_mind)
+		thralls += new_thrall_mind
+		new_thrall_mind.special_role = "thrall"
+		message_admins("[key_name_admin(new_thrall_mind.current)] was enthralled by a shadowling!")
+		log_game("[key_name(new_thrall_mind.current)] was enthralled by a shadowling!")
+
+		new_thrall_mind.AddSpell(new /obj/effect/proc_holder/spell/self/lesser_shadowling_hivemind(null))
+		new_thrall_mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/lesser_glare(null))
+		new_thrall_mind.AddSpell(new /obj/effect/proc_holder/spell/self/lesser_shadow_walk(null))
+		new_thrall_mind.AddSpell(new /obj/effect/proc_holder/spell/self/thrall_vision(null))
+		new_thrall_mind.current << "<span class='shadowling'><b>You see the truth. Reality has been torn away and you realize what a fool you've been.</b></span>"
```

Comment:
Stop looking at the PR's page, look directly at the branch. This has been fixed!!